client/state/ui/editor/reducer.js

```diff
@@ -33,6 +34,7 @@ export function postId( state = null, action ) {
 export default combineReducers( {
 	postId,
 	imageEditor,
+	videoEditor,
```

Comment:
The video editor is a modal that is opened from the media library modal. Media library modal handles passing the particular video and everything else the video editor needs to know. And because it's a modal, site switching should be a non-issue. Hope that helps clarify and makes sense.